From 78a678dfc0987c682ab7dd8e10759cc2761bb5f0 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Mon, 2 Oct 2017 23:27:53 +0000
Subject: [PATCH] TOOLS-1877 bin/crimport broken by Github API changes Reviewed
 by: Joshua M. Clulow <jmc@joyent.com> Approved by: Joshua M. Clulow
 <jmc@joyent.com>

---
 bin/crimport | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/bin/crimport b/bin/crimport
index 39f8826..1608ba2 100755
--- a/bin/crimport
+++ b/bin/crimport
@@ -255,7 +255,6 @@ function igp_github_protect
 
 	local TOKEN="$(cat $igp_github_token_file)"
 	local BRANCH="master"
-	local PREVIEW_HEADER="Accept: application/vnd.github.loki-preview+json"
 	local UNIQUE="$RANDOM$RANDOM$RANDOM:"
 
 	# Check that the Gerrit user is able to push
@@ -280,7 +279,7 @@ function igp_github_protect
 	fi
 
 	# Set branch protection on master, so that only Gerrit can push.
-	curl -fsS -u "$TOKEN:x-oauth-basic" -H "$PREVIEW_HEADER" -X PUT -d@- \
+	curl -fsS -u "$TOKEN:x-oauth-basic" -X PUT -d@- \
 	    "$igp_github_api/repos/$1/$2/branches/$BRANCH/protection" \
 	    > /dev/null <<-EOF
 		{
@@ -288,6 +287,14 @@ function igp_github_protect
 		        "strict": true,
 		        "contexts": [ ]
 		    },
+		    "required_pull_request_reviews": {
+		        "dismissal_restrictions": {
+		            "users": [],
+		            "teams": []
+		        },
+		        "dismiss_stale_reviews": true,
+		        "require_code_owner_reviews": true
+		    },
 		    "enforce_admins": true,
 		    "restrictions": {
 		        "users": [
-- 
2.21.0

